### ISSUE TITLE
test zot against s3 storage

### ISSUE THREAD

**Issue by andaaron** [anuvu/zot#210]
Theoretically zot should work with S3 storage if we:
- Use https://github.com/s3fs-fuse/s3fs-fuse to mount the S3 storage in the zot container.
- Use  https://github.com/anuvu/zot/commit/28974e81dcd449e78d4735f08b960b88f5ea068a

In order to test this integration
- Setup a proof of concept environment where minio + https://github.com/s3fs-fuse/s3fs-fuse are used
- Manually test the proof of concept environment.
- Test the performance with parallel workload mix - there are some sample tests here: https://github.com/anuvu/zot/commit/28974e81dcd449e78d4735f08b960b88f5ea068a#diff-be341d4347a3a838be38fb875f2c6b5e89bf5fb37921adb784636aa8d1935967R374
- We need to implement test scripts to test this s3 integration automatically, using GH actions.

In the production version of this feature, the S3 configuration parameters as well as the zot configuration would be passed to the zot container using kubernetes specific objects.

**Comment by rchincha** [comment]
Issue #206 
Issue #179

**Comment by peusebiu** [comment]
Small test with TestParallelRequests Suite:

normal one: 
```
--- PASS: TestParallelRequests (0.18s)
    --- PASS: TestParallelRequests/Request-4 (9.14s)
    --- PASS: TestParallelRequests/Request-2 (12.48s)
    --- PASS: TestParallelRequests/Request-6 (12.89s)
    --- PASS: TestParallelRequests/Request-12 (12.97s)
    --- PASS: TestParallelRequests/Request-8 (13.04s)
    --- PASS: TestParallelRequests/Request-3 (13.70s)
    --- PASS: TestParallelRequests/Request-1 (16.81s)
    --- PASS: TestParallelRequests/Request-5 (17.36s)
    --- PASS: TestParallelRequests/Request-10 (10.42s)
    --- PASS: TestParallelRequests/Request-14 (9.63s)
    --- PASS: TestParallelRequests/Request-11 (13.62s)
    --- PASS: TestParallelRequests/Request-9 (13.41s)
    --- PASS: TestParallelRequests/Request-13 (12.69s)
    --- PASS: TestParallelRequests/Request-7 (13.55s)
PASS
ok      github.com/anuvu/zot/pkg/api    26.668s
```

s3fuse + minio:

```
--- PASS: TestParallelRequests (0.29s)
    --- PASS: TestParallelRequests/Request-4 (39.80s)
    --- PASS: TestParallelRequests/Request-3 (76.65s)
    --- PASS: TestParallelRequests/Request-6 (78.74s)
    --- PASS: TestParallelRequests/Request-8 (78.74s)
    --- PASS: TestParallelRequests/Request-2 (79.31s)
    --- PASS: TestParallelRequests/Request-12 (79.31s)
    --- PASS: TestParallelRequests/Request-1 (152.00s)
    --- PASS: TestParallelRequests/Request-10 (75.82s)
    --- PASS: TestParallelRequests/Request-14 (74.84s)
    --- PASS: TestParallelRequests/Request-7 (157.56s)
    --- PASS: TestParallelRequests/Request-13 (84.86s)
    --- PASS: TestParallelRequests/Request-5 (143.05s)
    --- PASS: TestParallelRequests/Request-9 (105.13s)
    --- PASS: TestParallelRequests/Request-11 (106.02s)
PASS
ok      github.com/anuvu/zot/pkg/api    185.089s
```

**Comment by rchincha** [comment]
Maybe benchmarks also? https://pkg.go.dev/testing#hdr-Benchmarks
Hard to tell what the latency, etc is.
Wouldn't mind if that is part of #202.

**Comment by rchincha** [comment]
One good argument against using fuse mounts is that perhaps under some circumstances such as deploying containers, privileges to mount are not available/allowed. In which case, we may have to directly interact with the s3 APIs.

**Comment by peusebiu** [comment]
closed in 9c568c0ee26decab5b833e166637d54d28e58bb8
